server/util/clickhouse/clickhouse.go

```diff
@@ -210,7 +217,7 @@ func (h *DBHandle) FlushInvocationStats(ctx context.Context, ti *tables.Invocati
 		metrics.ClickhouseStatusLabel: statusLabel,
 	}).Inc()
 	if lastError != nil {
-		return status.UnavailableErrorf("clickhouse.FlushInvocationStats exceeded retries for invocation id %q, err: %s", ti.InvocationID, lastError)
+		return status.UnavailableErrorf("clickhouse.FlushInvocationStats exceeded retries (n=%d) for invocation id %q, err: %s", attempt, ti.InvocationID, lastError)
```

Comment:
the retrier keeps track of retry/attempt for you, you can call AttemptNumber and MaxAttempts to get this info

Reply:
Nice.